### PR TITLE
Documentation for S3 sync rules

### PR DESCRIPTION
**Pull request by akanshi-elastic**
#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes

### CODE REVIEW

File: docs/sync-rules/s3.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+### Example advanced sync rules
+
+#### Indexing files and folders recursively
+
+```json
+[
+  {
+    "bucket": "bucket1",
+    "prefix": "folder1/folder2", # this will fetch files starting with folder2 and files in folder2
```
[the remark this answers is not in the document]
Comment (navarone-feekery) [reply docs/sync-rules/s3.md:13]: As an example, if you look at our elastic docs, you can see that the key names are defined above the sample code, with detailed explanations on what they do. The sample code is then comment-free.
https://www.elastic.co/guide/en/elasticsearch/reference/8.8/indices-create-index.html

File: docs/sync-rules/s3.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+1. bucket: Name of Amazon S3 bucket for which the rule would execute.
+2. extension(optional): List of file extensions to be added for the sync. Defaults to empty list (i.e.sync all file extensions).
+3. prefix(optional): Prefix is a string of characters at the beginning of the object key name, this will index files and folder data that matches with prefix value. Defaults to ""(i.e. sync all bucket objects).
+
+### Example advanced sync rules
+
+### Indexing files and folders recursively by specifying prefix
+
+#### Case1: Fetch files/folder starting with docs. Use '/' to fetch from subdirectory
```
Comment (wangch079) [inline docs/sync-rules/s3.md:17]:
```suggestion
#### Case1: Fetch files/folder starting with docs in folder1. Use '/' to fetch from subdirectory
```

**Review by wangch079 — COMMENTED** [review]
cc @leemthompo

**Comment by leemthompo** [comment]
@wangch079 are you happy with the technical details here?

File: docs/sync-rules/s3.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+See the [Developer guide](../../docs/DEVELOPING.md) for setting up connectors.
+
+### Overview
+
+The advanced sync rules consist of sets of rules, each having one or more parameters:
+
+1. bucket: Name of Amazon S3 bucket for which the rule would execute.
+2. extension(optional): List of file extensions to be added for the sync. Defaults to empty list (i.e.sync all file extensions).
+3. prefix(optional): Prefix is a string of characters at the beginning of the object key name, this will index files and folder data that matches with prefix value. Defaults to ""(i.e. sync all bucket objects).
```
Comment (leemthompo) [inline docs/sync-rules/s3.md:11]:
```suggestion
Advanced Sync Rules help manage data in Amazon S3 buckets. They take the following parameters:

1. `bucket`: This is the Amazon S3 bucket the rule applies to.
2. `extension` (optional): Lists which file types to sync. Defaults to syncing all types.
3. `prefix` (optional): String of prefix characters. The connector will index files and folder data that matches the string. Defaults to `""` (i.e. sync all bucket objects).
```

File: docs/sync-rules/s3.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+### Example advanced sync rules
+
+### Indexing files and folders recursively by specifying prefix
```
Comment (leemthompo) [inline docs/sync-rules/s3.md:15]:
```suggestion
### Indexing files and folders recursively by prefix
```

File: docs/sync-rules/s3.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+#### Case1: Fetch files/folder starting with docs in folder1. Use '/' to fetch from subdirectory
```
Comment (leemthompo) [inline docs/sync-rules/s3.md:17]:
```suggestion
#### Example 
Fetch files/folders in `folder1/docs`.
```

File: docs/sync-rules/s3.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  {
+    "bucket": "bucket1",
+    "prefix": "folder1/docs"
+  }
+
+]
+
+```
+
+#### Case2: Fetch files/folder starting with folder1
```
Comment (leemthompo) [inline docs/sync-rules/s3.md:30]:
```suggestion
#### Example

Fetch files/folder starting with `folder1`.
```

File: docs/sync-rules/s3.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+### Overview
+
+The advanced sync rules consist of sets of rules, each having one or more parameters:
+
+1. bucket: Name of Amazon S3 bucket for which the rule would execute.
+2. extension(optional): List of file extensions to be added for the sync. Defaults to empty list (i.e.sync all file extensions).
+3. prefix(optional): Prefix is a string of characters at the beginning of the object key name, this will index files and folder data that matches with prefix value. Defaults to ""(i.e. sync all bucket objects).
+
+### Example advanced sync rules
```
Comment (leemthompo) [inline docs/sync-rules/s3.md:13]:
```suggestion
### Advanced sync rules examples
```

File: docs/sync-rules/s3.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    "bucket": "bucket2",
+    "prefix": "folder1"
+  }
+]
+
+```
+
+### Indexing files and folders by specifying extensions
+
+#### Case1: Fetch all objects which starts with `abc` and then filter using extension
```
Comment (leemthompo) [inline docs/sync-rules/s3.md:44]:
```suggestion
#### Example

Fetch all objects which start with `abc` and then filter using file extensions.
```

**Review by leemthompo — APPROVED** [review]
Some minor cleanup. Thanks!

**Review by leemthompo — APPROVED** [review]

**Pull request merged**